documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.md

```diff
@@ -0,0 +1,526 @@
+
+Although the `DeletedMedia` records are deleted, the ModerationDecisions
+preserve the history of these actions.
+
+### Other actions
+
+We do _not_ need to add any new action types for `rejected_reports` and
+`duplicate_reports`, which are only applicable to the Report view flow described
+in the related IP.
+
+## Views
+
+We will add one new read-only view for viewing ModerationDecisions, but most
+changes will be to existing views.
+
+### Media views
+
+We will add the custom bulk actions for `deindex_sensitive`,
+`deindex_copyright`, and `mark_sensitive` to the dropdown in the Media views.
+Bulk moderation decisions of these types will be created by taking these
+actions.
+
+However the current Media views only allow filtering records by uuid. To
+facilitate selecting records for bulk action, we'll update the `search_fields`
+for the main searchbar to search text fields (title, description, tags). We'll
+also add some additional text filters to the sidebar:
+
+- `uuid` - it may still be necessary to search by uuid, so we should preserve
+  this functionality. We include it in a separate filter (rather than keeping it
+  in `search_fields`) for search efficiency.
+- `provider`
+- `creator` - this field should have helptext reminding the user to also filter
+  by provider in order to disambiguate a specific creator.
+
+Additional or more sophisticated filters may be added in the future as
+appropriate.
+
+### DeletedMedia List View
+
+We'll add the `reversed_deindex` action to this list view, allowing maintainers
+to select a list of deindexed records and undo deindexing from this view.
+
+To facilitate easily undoing a particular bulk ModerationDecision, we will add a
+text filter for searching by ModerationDecision id. That allows you to select
+all records that were deleted as part of a particular decision.
+
+### SensitiveMedia List View
+
+We will add the `reversed_mark_sensitive` action to this list view, allowing
+maintainers to select a list of sensitive media and unmark them as sensitive
+from this view.
+
+Similar to the DeletedMedia list view, we will add a filter for
+ModerationDecision id to facilitate easily undoing an entire bulk
+ModerationDecision.
+
+### ModerationDecisions List View
+
+To make it easy to view historical ModerationDecisions, we will add one new list
+view. In the view, decisions will be sorted by creation date, and will display:
+
+- `id`
+- `action`
+- `explanation` - truncated `explanation` column
+- `record_count` - a count for the number of records affected by this decision
+
+This view is useful for getting the id of a moderation decision that you want to
+reverse. Clicking on an individual ModerationDecision will take you to a detail
+view that is entirely read-only, as a moderation decision should not be edited
+after it is created. We will add a custom filter to display only bulk moderation
+decisions, defined as a ModerationDecision with more than one media record.
```

Comment:
Would the single object view show a list of the works affected? Or the deleted/sensitive media records? That list would also probably need to be actionable, if it existed. I suppose the alternative is to go here, find the moderation decision you are after, then use it to filter the deleted/sensitive media views?

I'm thinking of a workflow where someone wants to easily reverse an entire decision, rather than a subset.

Maybe the deleted/sensitive list views should also include the moderation decision ID for each line as a clickable filter for that view? It would help in this case to not have to jump between the views.

I might be over complicating things though :sweat_smile:

Reply:
I think allowing reversal from the detail view introduces a ton of complexity (see [this comment](https://github.com/WordPress/openverse/pull/3719#discussion_r1473372955)), although that was definitely something I initially tried to plan for.

But we could definitely add some convenience links. What about adding a link to the Moderation Decision detail view that will take you to the deleted/sensitive list view (whichever is relevant), filtered by the moderation decision ID?

Reply:
> What about adding a link to the Moderation Decision detail view that will take you to the deleted/sensitive list view (whichever is relevant), filtered by the moderation decision ID?

Yeah, that sounds perfect!